.github/workflows/main.yml

```diff
@@ -178,7 +178,24 @@ jobs:
 
       - name: Run Clang-Tidy
         if: ${{ matrix.name == 'Clang-Tidy' }}
-        run: |
-          run-clang-tidy -j4 -quiet \
-            -checks="-*,clang-analyzer-*,-clang-analyzer-cplusplus*,-clang-analyzer-security*" \
-            -p build
+        uses: cpp-linter/cpp-linter-action@v2
+        id: cpp-linter
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          version: 18
+          step-summary: true
+          files-changed-only: false
+          style: ''  # disable clang-format checks.
+          tidy-checks: >-
+            -*
+            ,clang-analyzer-*
+            ,-clang-analyzer-cplusplus*
+            ,-clang-analyzer-security*
+            ,-clang-analyzer-valist*
+          database: 'build'
+          ignore: 'miniz|spng|win32|toolsrc|src/i_pcsound.c'
```

Comment:
Why do we exclude `src/i_pcsound.c`?

Reply:
i_pcsound.c only works if `HAVE_AL_BUFFER_CALLBACK` is defined (ubuntu-24.04 doesn't have it). Although clang-tidy supports a compiler database, it lacks any file filtering, so we have to filter it manually.